src/main/scala/util/RecordListMap.scala

```diff
@@ -0,0 +1,26 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.util
+
+import Chisel._
+import chisel3.Record
+import scala.collection.immutable.ListMap
+
+final case class RecordListMap[T <: Data](eltMap: ListMap[String, T])
+    extends Record with collection.Map[String, T] {
+
+  // Required by collection.Map[String, T]
+  def get(key: String) = eltMap.get(key)
+  def -(key: String) = eltMap - key
+  def +[V1 >: T](kv: (String, V1)) = eltMap + kv
+  def iterator = eltMap.iterator
```

Comment:
Also what do these mean in the context of `RecordListMap` extending `Data`? The existence of these methods imply some utility but they just return to you a `ListMap[String, T]` that means what exactly?

I think we should either return a new `RecordListMap` when calling these operations, or we should not extend `collection.Map`.

Reply:
to be clear, you're fine with `get` just returning `T`, but the `-` and `+` should return RecordListMap[T], right? what should the iterator type be?

Reply:
I updated the + and -, but not sure about the iterator

Reply:
nope, that was wrong. Struggling with the contravariance (?) required of the `+` operator. 

Should I just NOT extend the Map API?

Reply:
> Should I just NOT extend the Map API?

I think that would be the easiest thing to do, just implement whatever methods you think are useful.